### ISSUE TITLE
Overall task input size

### ISSUE THREAD

**Issue by cjllanwarne** [openwdl/wdl#225]
TL;DR: 
A new function to return the total size of all inputs to a task, eg
```wdl
task foo {
  File input1
  StructType input2
  Array[File] input3

  Float size_sum = total_input_size("GB")

  command { ... }
  runtime { ... }
  output { ... }
}
```

---

Detail:

It's a fairly regular pattern amongst people to want to base the disk/memory/cpu for a task on the size of input files. Where this gets tricky is that it's (a) tedious and (b) error-prone to have to include every input file individually in this calculation. Especially after a refactor, it's easy to accidentally forget to include one of the files into the sum or miss one because it's nested in an object/struct.

I'd like to propose a function `total_input_size()`, only available within `task` definitions, which would return the total size of every file needing to be localized to the execution environment for the task.

Like the current `size()` function, I'd include the optional unit parameter to let people specify the result in `MB`, `GiB`, etc.

Before I write up a SPEC change proposal does this sound like a good idea or do people have concerns?

**Comment by ruchim** [comment]
I think this will be a very useful utility! +1

**Comment by brendanreardon** [comment]
+1 This sounds wonderful!

**Comment by yfarjoun** [comment]
How will it interact with optional_localization?

**Comment by geoffjentry** [comment]
@yfarjoun That's not a WDL feature :)

**Comment by pshapiro4broad** [comment]
This is a good idea but it doesn't appear sufficient to handle cases we have in practice. Our WDLs often have different scaling applied to different inputs. E.g.
```
  Int disk_size = round(2 * (size(gvcf, "GB") + size(gvcf_index, "GB")) + 1)
```
and 
```
  Float compressed_data_size = size(funcotator_aou_data_source_tar_gz, "GB")
  Float ref_inputs_size = size(ref_fasta, "GB") + size(ref_fasta_index, "GB") + size(ref_dict, "GB")
[...]
        disk_space_gb = ceil(2 * vcf_inputs_size + 10 * compressed_data_size + ref_inputs_size) + 1,
```
etc

**Comment by cjllanwarne** [comment]
@yfarjoun as Jeff says, that's up to you - I'm pretty sure that evaluating `size(localization_optional_file)` is still the full file size, FWIW

**Comment by patmagee** [comment]
@cjllanwarne this looks like a good idea. I think I have even asked for this in the past 👍 .
@pshapiro4broad Im not sure if that is really an issue something like this would solve. There are MANY cases where just knowing the total disk space that will be taken up by localized files is very beneficial. Scaling of individual sizes is something you would do with the `size` function individually

**Comment by pshapiro4broad** [comment]
Sorry, I should've included a constructive suggestion instead of just a complaint.

What if there was a way to annotate inputs with a size scaling factor? I think that could work for us, but I don't know what the WDL way to do this would look like.

**Comment by vdauwera** [comment]
Would this basically be syntactic sugar to shortcut using size(Array[input Files]) as was added by #169?

**Comment by patmagee** [comment]
@vdauwera yes, that is exactly the case. I think we can actually close this in favor of #169
